shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -92,6 +92,19 @@ - (void)userNotificationCenter:(UNUserNotificationCenter*)center
   }
 }
 
+/**
+ * Calls all plugins registered for `UNUserNotificationCenterDelegate` callbacks.
+ */
+- (void)userNotificationCenter:(UNUserNotificationCenter*)center
+    didReceiveNotificationResponse:(UNNotificationResponse*)response
+             withCompletionHandler:(void (^)(void))completionHandler API_AVAILABLE(ios(10)) {
+  if (@available(iOS 10.0, *)) {
```

Comment:
I don't think you need this anymore - have you tested compilation on something targetting iOS 9?